File: 品味Java的21種設計模式/dp_java/templatemethod/example4/NormalLoginModel.java
package cn.javass.dp.templatemethod.example4;
/**
 * 封装进行登录控制所需要的数据，在公共数据的基础上，
 * 添加具体模块需要的数据
 */
public class NormalLoginModel extends LoginModel{
	/**
	 * 密码验证问题
	 */
	private String question;
	/**
	 * 密码验证答案
	 */
	private String answer;
	public String getQuestion() {
		return question;
	}
	public void setQuestion(String question) {
		this.question = question;
	}
	public String getAnswer() {
		return answer;
	}
	public void setAnswer(String answer) {
		this.answer = answer;
	}
	
}

File: 品味Java的21種設計模式/dp_cpp/templatemethod/example4/NormalLoginModel.cpp
#include "NormalLoginModel.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace templatemethod
			{
				namespace example4
				{

					std::string NormalLoginModel::getQuestion()
					{
						return question;
					}

					void NormalLoginModel::setQuestion(std::string question)
					{
						this->question = question;
					}

					std::string NormalLoginModel::getAnswer()
					{
						return answer;
					}

					void NormalLoginModel::setAnswer(std::string answer)
					{
						this->answer = answer;
					}
				}
			}
		}
	}
}